sentry_sdk/integrations/logging.py

```diff
@@ -139,7 +139,7 @@ def _extra_from_record(record):
     return {
         k: v
         for k, v in iteritems(vars(record))
-        if k not in COMMON_RECORD_ATTRS and not k.startswith("_")
+        if k not in COMMON_RECORD_ATTRS and not (isinstance(k, str) and k.startswith("_"))
```

Comment:
can you change this into `and (not k.startswith("_") or not isinstance(k, str)`? Not simpler but easier to read for me.

Reply:
this will lead to an AttributeError in `k.startswith("_")`

Reply:
sorry, of course the swapped variant `not isinstance(k, str) or not ...`

Reply:
Ok